api_server.go

```diff
@@ -278,13 +280,138 @@ func (x *ApiServer) SheetsUpload(w http.ResponseWriter, r *http.Request) {
 		}
 
 	default:
-		http.Error(w, "", http.StatusMethodNotAllowed)
+		methodNotAllowed(w)
+	}
+}
+
+type UploadType string
+
+func (x UploadType) String() string { return string(x) }
+
+const (
+	UploadTypeClickTrack UploadType = "click_track"
+	UploadTypeSheetMusic UploadType = "sheet_music"
+)
+
+type Upload struct {
```

Comment:
Might consider using protobuf here, if we decide to make requests via js.
We can accept both json and protobuf content types, but for _our_ js client, we'll opt for protobuf.

Reply:
We can also use enums for `UploadTypeClickTrack` and `UploadTypeSheetMusic` if we go with protobuf.